libraries/exoplayer_rtsp/src/test/java/androidx/media3/exoplayer/rtsp/reader/RtpH263ReaderTest.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package androidx.media3.exoplayer.rtsp.reader;
+
+import static androidx.media3.common.util.Util.getBytesFromHexString;
+import static com.google.common.truth.Truth.assertThat;
+
+import androidx.media3.common.Format;
+import androidx.media3.common.MimeTypes;
+import androidx.media3.common.util.ParsableByteArray;
+import androidx.media3.exoplayer.rtsp.RtpPacket;
+import androidx.media3.exoplayer.rtsp.RtpPayloadFormat;
+import androidx.media3.test.utils.FakeExtractorOutput;
+import androidx.media3.test.utils.FakeTrackOutput;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.primitives.Bytes;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+/**
+ * Unit test for {@link RtpH263Reader}.
+ */
+@RunWith(AndroidJUnit4.class)
+public final class RtpH263ReaderTest {
+  private static final byte[] FRAME_1_FRAGMENT_1_DATA =
+      getBytesFromHexString("80020c0419b7b7d9591f03023e0c37b");
+  private final RtpPacket FRAME_1_FRAGMENT_1 =
+      new RtpPacket.Builder()
+          .setTimestamp((int) 2599168056L)
+          .setSequenceNumber(40289)
+          .setMarker(false)
+          .setPayloadData(
+              Bytes.concat(
+                  /*payload header */ getBytesFromHexString("0400"), FRAME_1_FRAGMENT_1_DATA))
+          .build();
+  private static final byte[] FRAME_1_FRAGMENT_2_DATA =
+      getBytesFromHexString("03140e0e77d5e83021a0c37");
+  private static final RtpPacket FRAME_1_FRAGMENT_2 =
+      new RtpPacket.Builder()
+          .setTimestamp((int) 2599168056L)
+          .setSequenceNumber(40290)
+          .setMarker(true)
+          .setPayloadData(
+              Bytes.concat(
+                  /*payload header */ getBytesFromHexString("0000"), FRAME_1_FRAGMENT_2_DATA))
+          .build();
+  private static final byte[] FRAME_1_DATA =
+      Bytes.concat(getBytesFromHexString("0000"), FRAME_1_FRAGMENT_1_DATA, FRAME_1_FRAGMENT_2_DATA);
```

Comment:
curious, why the data needs `0000` in the beginning?

Reply:
According to RFC4629 Section 6.1.1.  " A packet that begins at the location of a Picture, GOB, slice, EOS,
or EOSBS start code shall omit the first two (all zero) bytes from the H.263+ bitstream", H263 reader changes the payload header data (i.e 0400) to 0000. So the final output contains 0000 + frame data.